rskj-core/src/main/java/co/rsk/peg/BridgeSupportFactory.java

```diff
@@ -0,0 +1,59 @@
+package co.rsk.peg;
+
+import co.rsk.bitcoinj.core.Context;
+import co.rsk.config.BridgeConstants;
+import co.rsk.core.RskAddress;
+import co.rsk.peg.BtcBlockStoreWithCache.Factory;
+import co.rsk.peg.utils.BridgeEventLogger;
+import co.rsk.peg.utils.BridgeEventLoggerImpl;
+import java.util.List;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.core.Block;
+import org.ethereum.core.Repository;
+import org.ethereum.vm.LogInfo;
+import org.ethereum.vm.PrecompiledContracts;
+
+/**
+ * BridgeSupportFactory allows BridgeSupport instantiation.
+ */
+public class BridgeSupportFactory {
+
+    private Factory btcBlockStoreFactory;
+    private BridgeConstants bridgeConstants;
+    private ActivationConfig activationConfig;
+
+    public BridgeSupportFactory(Factory btcBlockStoreFactory,
+            BridgeConstants bridgeConstants, ActivationConfig activationConfig) {
+
+        this.btcBlockStoreFactory = btcBlockStoreFactory;
+        this.bridgeConstants = bridgeConstants;
+        this.activationConfig = activationConfig;
+    }
+
+
+    public BridgeSupport newInstance(Repository repository, Block executionBlock,
+            RskAddress contractAddress, List<LogInfo> logs) {
+
+        ActivationConfig.ForBlock activations = activationConfig.forBlock(executionBlock.getNumber());
+        Context btcContext = new Context(bridgeConstants.getBtcParams());
+
+        BridgeStorageProvider provider = new BridgeStorageProvider(
+            repository,
+            contractAddress,
+            bridgeConstants,
+            BridgeStorageConfiguration.fromBlockchainConfig(activations)
+        );
+
+        FederationSupport federationSupport = new FederationSupport(bridgeConstants, provider, executionBlock);
+
+        BridgeEventLogger eventLogger;
+        if (logs == null) {
```

Comment:
Where does this case come from?

Reply:
EthModule doesn't provide logs when creating the BridgeSupport instance. It might require a bigger refactor to avoid this.